Jucer2Reprojucer/main.cpp

```diff
@@ -498,19 +498,52 @@ int main(int argc, char* argv[])
     if (projectType == "audioplug")
     {
       wLn("jucer_audio_plugin_settings(");
-      convertOnOffSettingWithDefault(jucerProject, "buildVST", "BUILD_VST", true);
-      convertOnOffSettingWithDefault(jucerProject, "buildVST3", "BUILD_VST3", false);
-      convertOnOffSettingWithDefault(jucerProject, "buildAU", "BUILD_AUDIOUNIT", true);
-      convertOnOffSettingWithDefault(jucerProject, "buildAUv3", "BUILD_AUDIOUNIT_V3",
-                                     false);
-      convertOnOffSettingWithDefault(jucerProject, "buildRTAS", "BUILD_RTAS", false);
-      convertOnOffSettingWithDefault(jucerProject, "buildAAX", "BUILD_AAX", false);
-      if (jucerVersionAsTuple >= Version{5, 0, 0})
+
+      if (jucerVersionAsTuple >= Version{5, 3, 1})
       {
-        convertOnOffSettingWithDefault(jucerProject, "buildStandalone",
-                                       "BUILD_STANDALONE_PLUGIN", false);
-        convertOnOffSettingWithDefault(jucerProject, "enableIAA", "ENABLE_INTERAPP_AUDIO",
+        const auto convertVarsToValues =
+          [](const juce::var& v, const juce::StringArray& allVars,
+             const juce::StringArray& allValues) -> juce::StringArray {
+          auto vars = juce::StringArray::fromTokens(v.toString(), ",", {});
+          juce::StringArray values;
+          for (auto i = 0; i < allValues.size(); ++i)
+          {
+            if (vars.contains(allVars[i]))
+            {
+              values.add(allValues[i]);
+            }
+          }
+          return values;
+        };
+
+        const juce::StringArray formatsVars = {
```

Comment:
This is pretty difficult to read. Why not put them in the lambda, or name the lamba variables like the callsite ?

Reply:
s/allVars/formatVars/
s/allValues/formatValues/

Reply:
This would be more straighforward if written as a translation list:

```cpp
const auto convertVarsToValues =
    [](const juce::var& v, const std::vector<std::pair<juce::String, juce::String>>& varToValues) -> juce::StringArray {
    auto vars = juce::StringArray::fromTokens(v.toString(), ",", {});
    juce::StringArray values;
    for (const auto& [ var, value ] : varToValues)
        if(vars.contains(var)) {
            values.add(value);              
       }
    }
    return values;
};
const auto varToValues = {
    {"buildVST", "VST"},  
    {"buildVST3", "VST3"},
// etc.
};
convertSettingAsList(jucerProject, "pluginFormats", "PLUGIN_FORMATS",
                             [&](const juce::var& v) {
                               return convertVarsToValues(v, varToValues);
                             });
```